sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ColumnResolutionHelper.scala

```diff
@@ -315,32 +317,34 @@ trait ColumnResolutionHelper extends Logging with DataTypeErrorsBase {
       }.map(e => Alias(e, nameParts.last)())
     }
 
-    def innerResolve(e: Expression, isTopLevel: Boolean): Expression = withOrigin(e.origin) {
+    def innerResolve(e: Expression, isTopLevel: Boolean): Expression = {
       if (e.resolved || !e.containsAnyPattern(UNRESOLVED_ATTRIBUTE, TEMP_RESOLVED_COLUMN)) return e
```

Comment:
I think that Iv'e introduced a subtle bug here due to non-local return behavior: previously, the return was nested inside of a closure passed to `withOrigin` and returned from that closure, but now the return is returning from the _outer_ method which contains `innerResolve` rather than returning from `innerResolve` itself. 

An earlier version that I tested locally had used a return-free if-else pattern with additional braces to make the pattern clearer. Let me revert back to that here.